src/history/hs.c

```diff
@@ -53,32 +60,63 @@ __hs_store_time_pair(WT_SESSION_IMPL *session, wt_timestamp_t timestamp, uint64_
 int
 __wt_hs_config(WT_SESSION_IMPL *session, const char **cfg)
 {
+    WT_BTREE *btree;
     WT_CONFIG_ITEM cval;
-    WT_CURSOR_BTREE *hs_cursor;
-    WT_SESSION_IMPL *hs_session;
+    WT_CONNECTION_IMPL *conn;
+    WT_DECL_RET;
+    bool reset_no_dh_flag;
 
-    WT_RET(__wt_config_gets(session, cfg, "history_store.file_max", &cval));
+    conn = S2C(session);
+    reset_no_dh_flag = false;
 
+    WT_ERR(__wt_config_gets(session, cfg, "history_store.file_max", &cval));
     if (cval.val != 0 && cval.val < WT_HS_FILE_MIN)
-        WT_RET_MSG(session, EINVAL, "max history store size %" PRId64 " below minimum %d", cval.val,
+        WT_ERR_MSG(session, EINVAL, "max history store size %" PRId64 " below minimum %d", cval.val,
           WT_HS_FILE_MIN);
 
-    /* This is expected for in-memory configurations. */
-    hs_session = S2C(session)->cache->hs_session[0];
-    WT_ASSERT(session, hs_session != NULL || F_ISSET(S2C(session), WT_CONN_IN_MEMORY));
-
-    if (hs_session == NULL)
+    /* in-memory or readonly configurations do not have a history store. */
+    if (F_ISSET(conn, WT_CONN_IN_MEMORY | WT_CONN_READONLY))
         return (0);
 
+    /*
+     * Retrieve the btree from the history store cursor. This function might need to open a cursor
+     * from the default session and hence need to flip the no-dhandle flag temporarily in case it is
+     * set.
```

Comment:
@sulabhM can you add a FIXME here with a (new) linked ticket - I don't think the default session should ever open a history store cursor, but don't want to solve that right now.

Reply:
Added a TODO WT-5501